server/src/main/java/org/apache/cassandra/sidecar/handlers/OpenApiHandler.java

```diff
@@ -0,0 +1,246 @@
+            "../server/build/generated/openapi/" + fileName, // Alternative relative path
+        };
+        
+        Path foundPath = null;
+        for (String pathStr : possiblePaths)
+        {
+            Path path = Paths.get(pathStr);
+            if (Files.exists(path))
+            {
+                foundPath = path;
+                break;
+            }
+        }
+        
+        if (foundPath == null)
+        {
+            throw new IOException("Generated OpenAPI specification not found in classpath resources or file system. " +
+                                "Please run './gradlew openApiGenerate' and ensure the generated file is included in the build.");
+        }
+        
+        // Check if we need to reload the file (if it's newer than our cached version)
+        long fileLastModified = Files.getLastModifiedTime(foundPath).toMillis();
+        String cached = isYaml ? cachedYamlSpec : cachedJsonSpec;
+        if (cached == null || fileLastModified > lastModified)
+        {
+            logger.debug("Loading OpenAPI specification from: {}", foundPath.toAbsolutePath());
+            String content = Files.readString(foundPath, StandardCharsets.UTF_8);
+            lastModified = fileLastModified;
+            logger.info("Loaded OpenAPI specification from {}", foundPath.toAbsolutePath());
+            
+            // Update the appropriate cache
+            if (isYaml)
+            {
+                cachedYamlSpec = content;
+            }
+            else
+            {
+                cachedJsonSpec = content;
+            }
+            
+            return content;
+        }
+        
+        return cached;
+    }
+    
+    /**
+     * Creates a basic OpenAPI configuration as fallback when generated spec is not available
+     */
+    private static OpenAPI createOpenApiFromConfig(OpenApiConfiguration config)
+    {
+        OpenAPI openApi = new OpenAPI();
+        
+        // Set basic info
+        io.swagger.v3.oas.models.info.Info info = new io.swagger.v3.oas.models.info.Info();
+        info.setTitle(config.title());
+        info.setDescription(config.description() + " (Generated specification not available - run './gradlew openApiGenerate')");
+        info.setVersion(config.version());
+        
+        // Set license info
+        io.swagger.v3.oas.models.info.License license = new io.swagger.v3.oas.models.info.License();
+        license.setName(config.licenseName());
+        license.setUrl(config.licenseUrl());
+        info.setLicense(license);
+        
+        openApi.setInfo(info);
+        
+        // Set server info
+        io.swagger.v3.oas.models.servers.Server server = new io.swagger.v3.oas.models.servers.Server();
+        server.setUrl(config.serverUrl());
+        server.setDescription(config.serverDescription());
+        openApi.setServers(Collections.singletonList(server));
+        
+        return openApi;
+    }
```

Comment:
There is no use of this fallback. Can we just return a 404 if no spec is available?